components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -116,23 +139,11 @@ open class ExperimentsInternalAPI internal constructor() {
      * saves them to local storage
      */
     @Synchronized
-    internal fun updateExperiments(): Boolean {
-        if (!experimentsLoaded) {
-            loadExperiments()
-        }
+    internal fun onExperimentsUpdated(serverState: ExperimentsSnapshot) {
+        assert(experimentsLoaded)
```

Comment:
`assert` in Kotlin is [only enabled](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/assert.html) if "runtime assertions have been enabled on the JVM using the -ea JVM option.". This is usually the case in tests, as far as I can tell. If you want this code to crash with an useful message at runtime, you can use [`check`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/check.html).

Reply:
This is what i expect here. The intent is to have issues show up earlier and more clear in testing.

I don't know if that's the best pattern though, so i'm open to feedback on this.

Reply:
If it's testing, then this is fine as it is. I might suggest adding an assertion message as, sometimes, we're unable to get the file/line of the assertions on CI. I'm not sure why that happens, but it happens.

Reply:
Good point, ok.